server/src/test/java/org/elasticsearch/action/admin/cluster/storedscripts/ScriptMethodInfoSerializingTests.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.cluster.storedscripts;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.script.ScriptContextInfo.ScriptMethodInfo;
+import org.elasticsearch.test.AbstractSerializingTestCase;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ScriptMethodInfoSerializingTests extends AbstractSerializingTestCase<ScriptMethodInfo> {
+    private static String execute = "execute";
+    private static String getPrefix = "get";
+    private static int minLength = 1;
+    private static int maxLength = 16;
+
+    enum NameType {
+        EXECUTE,
```

Comment:
Not for now, but for the future, I wonder if it would be worth having needs methods here. It might be interesting to show these in our editor in the future based on what the user has done. For example, if they were to use _score in a score context we could show that needs for this is true, possibly highlighting different ramifications.

Reply:
Interesting thought.  When we get into developing the editor, I'll keep that in mind.

Reply:
My 2 cents: I'm not sure the needs methods are useful to users. They are an implementation detail of how internal consumers of the scripting infrastructure decide what, if any, additional setup is needed before executing the script. A user shouldn't ever care about them, and they are not on the script instance themselves, but on the factory, which is also something outside the scope of this api as it has to do with how scripts are constructed internally, while this api is about what is available to scripts when executing.